### PR TITLE
Enable ReceiveResourceSpansV2 by default in OTel Agent

### PR DESCRIPTION
**Pull request by IbraheemA**
### What does this PR do?

Set `enable_receive_resource_spans_v2` flag by default in Converged Agent. This enables the refactor of the span receiver in trace agent, from https://github.com/DataDog/datadog-agent/pull/29435.

Also, honour the `DD_APM_FEATURES` env variable and the `apm_config.features` core config option if either of them is set. The former takes precedence over the latter. 

### Describe how to test/QA your changes

1. Run the otel agent with no APM features set, and verify that the V2 Receiver logic is correctly used (e.g. "resource-related values (container, env, hostname) [aren't checked for] in span attributes")
     - e.g., send a span with `deployment.environment` set in span attributes and not in resource attributes. It should not get set in "env" in the output log in DD

2. Run the otel agent with `DD_APM_FEATURES=""` and verify that the V1 receiver logic is used.

### Possible Drawbacks / Trade-offs

Users may be trying to set other DD_APM_FEATURES (e.g. they set `DD_APM_FEATURES=sql_cache`) and disable enable_receive_resource_spans_v2 without realizing. Open to suggestions for how we should address this while retaining the option for users to disable `enable_receive_resource_spans_v2` intentionally.

### CODE REVIEW

File: cmd/otel-agent/config/agent_config.go [hunk 1]
```diff
@@ -176,10 +177,38 @@ func NewConfigComponent(ctx context.Context, ddCfg string, uris []string) (confi
 	if addr := ddc.Traces.Endpoint; addr != "" {
 		pkgconfig.Set("apm_config.apm_dd_url", addr, pkgconfigmodel.SourceFile)
 	}
-	if ddc.Traces.ComputeTopLevelBySpanKind {
-		pkgconfig.Set("apm_config.features", []string{"enable_otlp_compute_top_level_by_span_kind"}, pkgconfigmodel.SourceFile)
+
+	if apmConfigFeaturesEnv, found := os.LookupEnv("DD_APM_FEATURES"); found {
+		pkgconfig.Set("apm_config.features", strings.Split(apmConfigFeaturesEnv, ","), pkgconfigmodel.SourceEnvVar)
```
Comment (dineshg13) [inline cmd/otel-agent/config/agent_config.go:182]: Check for `pkgconfig.Get("apm_config.features") != nil ` it should suffice . It should read from the env var as well.

File: cmd/otel-agent/config/agent_config.go [hunk 2]
```diff
@@ -176,10 +177,38 @@ func NewConfigComponent(ctx context.Context, ddCfg string, uris []string) (confi
...
+						if fstr, ok := feature.(string); ok {
+							featuresStrings = append(featuresStrings, fstr)
+						} else {
+							success = false
+							break
+						}
+					}
+					if success {
+						pkgconfig.Set("apm_config.features", featuresStrings, pkgconfigmodel.SourceFile)
+					}
```
Comment (dineshg13) [inline cmd/otel-agent/config/agent_config.go:199]: it seems complicated and hard to follow. Can we simplify this

**Review by dineshg13 — CHANGES_REQUESTED** [review]

**Review by jhgilbert — APPROVED** [review]
Approved with minor suggestions for style, thank you!

**Comment by IbraheemA** [comment]
@jhgilbert did you mean to add comments? I can't see any

**Review by dineshg13 — APPROVED** [review]
LGTM. Can you please add how to verify that v2 is running in the OTel Agent
